src/status.js

```diff
@@ -254,62 +263,268 @@
 		return data;
 	};
 
+	Status.prototype._subscribeToMultidayStatusUpdate = function (robot_objects, callback) {
+		Logger.debug(`Subscribe to MultidayStatusUpdate`)
+		let subs = this.selector.subscribe({
+				service: 'status',
+				func: 'MultidayStatusUpdated',
+				obj: {
+					interface: 'fr.partnering.Status',
+					path: "/fr/partnering/Status"
+				}
+			}, (peerId, err, data) => {
+				if (err != null) {
+					Logger.error("Multiday status subscription: error", err)
+					return
+				}
+				if (!Array.isArray(data)) {
+					Logger.warn("Multiday status subscription: malformed data", data)
+					return
+				}
+				let robotToStatusMap = this._unpackRobotModels(data[0])
+				Logger.debug(`Multiday status subscription data after unpacking:`, robotToStatusMap)
+				for (var [key, value] of robotToStatusMap.entries()) {
+					let robotIds = key.split(':')
+					let robotId = robotIds[0]
+					let robotName = robotIds[1]
+					this._getRobotModelFromRecv2(value, robotId, robotName) // update this.robotModel
+				  }
+				Logger.debug(`RobotModel after unpacking:`, this.robotModel)
+				if (typeof callback === 'function') {
+					callback(this.robotModel);
+				}
+		})
+		this.subscriptions.push(subs)
+
+		Logger.debug(`Triggering MultidayStatusUpdate for robots`, robotNames)
+		let robotNames = robot_objects.map(robot => robot.RobotName)
+		this.selector.request({
+				service: "status",
+				func: "TriggerMultidayStatuses",
+				obj: {
+					interface: 'fr.partnering.Status',
+					path: "/fr/partnering/Status"
+				},
+				data: {
+					robot_names: robotNames
+				}
+			}, (peerId, err, data) => {
+				// Do nothing since the server should reponse back via signals
+				if (err != null) {
+					Logger.warn(`Multiday status trigger: error`, err)
+					if (typeof callback === 'function') callback(-1);
+					throw new Error(err)
+				}
+			})
+	}
+
+	/**
+	 * Get 'Parts' reference map to reduce status payload. Duplicated contents in status are stored in the map.
```

Comment:
It seems interesting.
However, we must assume that the information may not change in between because this is only one request. Probably acceptable for now.